xaitk_saliency/utils/detection.py

```diff
@@ -0,0 +1,56 @@
+from typing import Optional
+
+import numpy as np
+
+
+def format_detection(
+    bbox_mat: np.ndarray,
+    classification_mat: np.ndarray,
+    objectness: Optional[np.ndarray] = None,
+) -> np.ndarray:
+    """
+    Combine detection and classification output, with optional objectness
+    output, into the combined format required for
+    :py:meth:`.GenerateDetectorProposalSaliency.generate` ``*_dets`` input
+    parameters.
+
+    We enforce some shape consistency so that we can create a valid output
+    matrix. The input bounding box matrix should be of shape `[nDets x 4]`, the
+    classification matrix should be of shape `[nDets x nClasses]`, and the
+    objectness vector, if provided, should be of size `nDets`.
+
+    If an `objectness` score vector is not provided, we assume a vector of 1's.
+
+    The output of this function is a matrix that is of shape
+    `[nDets x (4+1+nClasses)]`. This is the result of horizontally stacking the
+    input in bbox, objectness and classification order. The output matrix
+    data-type will follow numpy's rules about safe-casting given the
+    combination of input matrix types.
+
+    In exceptions about shape mismatches, index 0 refers to the `bbox_mat`
+    input, index 1 refers to the ``objectness` vector, and index 2 refers to
```

Comment:
I will argue that this is not a typo as this get's rendered as RST content, and the intent here is to mark this word as *literals*, which uses double-backticks, where as the other content here that uses a single backtick are rendered as italicized (except for domain syntax, like `:py:meth:`, that is explicitly supposed to use single-backticks). If this is ever rendered as markdown, markdown rendered double-backticked content as literals as well in addition to the markdown-usual single-backticks.

I will grant you that I could correct the other occurrences here of the ticks so that we are consistent.

Reply:
So...?
```suggestion
    input, index 1 refers to the ``objectness`` vector, and index 2 refers to
```

Reply:
Yup, I will have an updated fixing this in a moment.